python/caffe/layers/detectnet/clustering.py

```diff
@@ -0,0 +1,215 @@
+import math
+
+import cv2 as cv
+import numpy as np
+
+import caffe
+
+MAX_BOXES = 50
+
+class ClusterGroundtruth(caffe.Layer):
+    """
+    * converts ground truth labels from grid format to list
+    * bottom[0] - coverage-label [ batch_size x grid_sz_x x grid_sz_y x 1 ]
+    * bottom[1] - bbox-label [batch_size x grid_sz_x x grid_sz_y x 4 (xl, yt, xr, yb)]
+    * top [0] - list of groundtruth bbox
+        [ batch_size x max_bbox_per_image x 5 (xl, yt, xr, yb, 0)]
+
+    Example prototxt definition:
+
+    layer {
+        type: 'Python'
+        name: 'cluster_gt'
+        # gt_bbox_list is a batch_size x MAX_BOXES x 5 blob
+        top: 'gt_bbox_list'
+        bottom: 'coverage-label'
+        bottom: 'bbox-label'
+        python_param {
+            module: 'caffe.layers.detectnet.clustering'
+            layer: 'ClusterGroundtruth'
+            # parameters - img_size_x, img_size_y, stride,
+            param_str : '1248,352,16'
+        }
+        include: { phase: TEST }
+    }
+    """
+
+    def setup(self, bottom, top):
+        self.is_groundtruth = True
+        try:
+            plist = self.param_str.split(',')
```

Comment:
Is `self.param_str` accessible? I didn't include those changes to `python/cafffe/_caffe.cpp`. This should be throwing an error...